contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+        require(_fee > 0, "OTCWrapper: fee cannot be 0");
+
+        fee[_underlying] = _fee;
+    }
+
+    /**
+     * @notice sets the beneficiary address
+     * @dev can only be called by owner
+     * @param _beneficiary beneficiary address
+     */
+    function setBeneficiary(address _beneficiary) external onlyOwner {
+        require(_beneficiary != address(0), "OTCWrapper: beneficiary address cannot be 0");
+
+        beneficiary = _beneficiary;
+    }
+
+    /**
+     * @notice sets the fill deadline duration
+     * @dev can only be called by owner
+     * @param _fillDeadline fill deadline duration
+     */
+    function setFillDeadline(uint256 _fillDeadline) external onlyOwner {
+        require(_fillDeadline > 0, "OTCWrapper: fill deadline cannot be 0");
+
+        fillDeadline = _fillDeadline;
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice allows market maker to deposit collateral
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID id of the order
+     * @param _amount amount to deposit
+     */
+    function depositCollateral(uint256 _orderID, uint256 _amount) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.seller == msg.sender, "OTCWrapper: sender is not the order seller");
+
+        IERC20 collateralInterface = IERC20(order.collateral);
+
+        // market maker inflow - an approve() by the msg.sender is required beforehand
+        collateralInterface.safeTransferFrom(msg.sender, address(this), _amount);
+
+        // approve margin pool to deposit collateral
+        collateralInterface.safeApproveNonCompliant(addressbook.getMarginPool(), _amount);
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.DepositCollateral,
+            address(this), // owner
+            address(this), // address to transfer from
+            order.collateral, // deposited asset
+            order.vaultID, // vaultId
+            _amount, // amount
+            0, //index
+            "" //data
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        emit CollateralDeposited(_orderID, _amount);
+    }
+
+    /**
+     * @notice allows market maker to withdraw collateral
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID id of the order
+     * @param _amount amount to withdraw
+     */
+    function withdrawCollateral(uint256 _orderID, uint256 _amount) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
```

Comment:
same as [this](https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/files#r1136446876) comment

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/f6866eef4da54cca9b8c6ee51c0e10b435284aa4